test/core.jl

```diff
@@ -92,6 +92,14 @@ let abcd = ABCDconst(1, 2, 3, 4)
     @test (1, 2, "not constant", 4) === (abcd.a, abcd.b, abcd.c, abcd.d)
 end
 
+# test `===` handling null pointer in struct #44712
+struct N44712
+    a::Some{Any}
+    b::Int
+    AB() = new()
+end
+
+@test unsafe_load(Ptr{N44712}(pointer(Int[0,1]))) !== unsafe_load(Ptr{N44712}(pointer(Int[0,2])))
```

Comment:
our unsafe test might need a little less danger
```suggestion
    N44712() = new()
end
let a  = Int[0, 1], b = Int[0, 2]
    GC.@preserve a b begin
        @test unsafe_load(Ptr{N44712}(pointer(a))) !== unsafe_load(Ptr{N44712}(pointer(b)))
    end
end
```